### PR TITLE
watch: new package

### PR DESCRIPTION
**Pull request by whitfin**
This PR adds the `watch` package, which is popular on Unix systems for repeating commands on interval.

### CODE REVIEW

File: var/spack/repos/builtin/packages/package.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Watch(Package):
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/package.py]: This should be converted to use the `AutotoolsPackage` base class.

File: var/spack/repos/builtin/packages/package.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    depends_on('libtool',      type='build')
+    depends_on('pkgconfig',    type='build')
+    depends_on('xz',           type='build')
+
+    depends_on('gettext')
+
+    phases = ['autogen', 'install']
+
+    def autogen(self, spec, prefix):
+        autogen = Executable('./autogen.sh')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/package.py]: Is this required? If we simply use the `configure` script that comes with the tarball, we can drop a few of those Autotools dependencies.

File: var/spack/repos/builtin/packages/package.py (hunk 1 above)
Comment (whitfin) [reply var/spack/repos/builtin/packages/package.py]: I was trying to get that to work, but how can I override the fact that it does a `make install` in that package?

Comment (ax3l) [reply var/spack/repos/builtin/packages/package.py]: If you just override the `install` method as you did, it should skip the `make install` default implementation and take yours :)

**Comment by chuckatkins** [comment]
Given that watch is just one small utility out of the proc-ps package, wouldn't it be better to have the whole proc-ps package instead of carving out pieces of it?

**Comment by whitfin** [comment]
@chuckatkins that's a fair point; my assumption was that it was just be largely complicated because you'd probably want to be able to opt in and out of each utility. For example, on my system I already have several of their utilities provided out of the box and so would not want probably half a dozen of them.

**Comment by adamjstewart** [comment]
Personally I would be fine with having packages for both `proc-ps` and `watch` if they are distributed separately. For example, if a package has a dependency on one part of `proc-ps`, it would be faster to only build that dependency if that's the only thing that's needed.

File: var/spack/repos/builtin/packages/watch/package.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    build_targets = ['watch']
+
+    def configure_args(self):
+        return [
+            '--disable-dependency-tracking',
+            '--disable-nls'
+        ]
+
+    def install(self, spec, prefix):
+        make('watch')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/watch/package.py]: This line isn't needed, it's taken care of during the `build` phase.

**Comment by whitfin** [comment]
@adamjstewart I have switched to Autotools, but I'm having difficulty because there's no `make install` in the utility (that I can tell). Is the `install` phase correct, or would you suggest another way?

**Comment by adamjstewart** [comment]
> Personally I would be fine with having packages for both `procps` and `watch` if they are distributed separately.

Hmm, they aren't distributed separately. Does building all of `procps` add any extra dependencies? How long is the build time? Maybe we should just have a `procps` package.

**Comment by whitfin** [comment]
It looks like their dependency listing is in their CI setup: https://gitlab.com/procps-ng/procps/blob/master/.gitlab-ci.yml. I haven't yet tried the build time of the entire thing; I only use `watch` :) 

Edit: also updated to remove that line.

**Comment by adamjstewart** [comment]
Oops, I completely forgot about this PR and accidentally wrote my own packages for `watch` and `procps` in #13044 and #13043. Sorry for the confusion, I didn't mean to steal your package. Let me know if my `watch` package works for you!

**Pull request closed without merging**